src/components/MapView.js

```diff
@@ -0,0 +1,427 @@
+import React from 'react';
```

Comment:
So is this intended to be a Presentational or Container Component? https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

Cause right now it appears to be trying to act as both.  Suggest refactoring, as it'll help with reuse down the road.

Reply:
yeah I agree, it's gotten bloated